packages/neo-one-client/src/user/LocalUserAccountProvider.ts

```diff
@@ -823,6 +841,172 @@ export class LocalUserAccountProvider<TKeyStore extends KeyStore, TProvider exte
+      .filter(
+        (output) =>
+          (looseAssets[common.uInt256ToString(output.asset)] as ReadonlyArray<UnspentOutput> | undefined) !==
+            undefined && common.uInt160ToString(output.address) === addressToScriptHash(from.address),
+      )
+      .reduce(
+        (acc, output) => {
+          const currentAsset = common.uInt256ToString(output.asset);
+          const toAdd = Math.min(
+            looseAssets[currentAsset].length,
+            Math.floor((byteLimit - getMessageSize(acc.newInputs.length, 0)) / newInputSize),
+          );
+          if (toAdd === 0) {
+            return acc;
+          }
+          const tempIns = acc.looserAssets[currentAsset]
+            .sort((coinA, coinB) => coinA.value.comparedTo(coinB.value))
+            .slice(0, toAdd);
+          const newUpdatedOutputs = acc.updatedOutputs.map((out) => {
+            if (out.address === output.address && out.asset === output.asset) {
+              return updateOutput(out, tempIns.reduce((left, right) => left.plus(right.value), new BigNumber('0')));
+            }
+
+            return out;
+          });
+          const evenLooserAssets = {
+            ...acc.looserAssets,
+            [currentAsset]: acc.looserAssets[currentAsset].slice(toAdd),
+          };
+
+          return {
+            newInputs: acc.newInputs.concat(this.convertInputs(tempIns)),
+            updatedOutputs: newUpdatedOutputs,
+            looserAssets: evenLooserAssets,
+          };
+        },
+        { newInputs: inputArray(), updatedOutputs: transactionUnsignedIn.outputs, looserAssets: looseAssets },
+      );
+
+    const { newestInputs, newOutputs } = Object.keys(looserAssets)
+      .filter((key) => looserAssets[key].length >= 2)
+      .reduce(
+        (acc, key) => {
+          if (byteLimit - getMessageSize(acc.newestInputs.length + 2, acc.newOutputs.length + 1) < 0) {
+            return acc;
+          }
+          const newUnspents = looserAssets[key]
+            .sort((coinA, coinB) => coinA.value.comparedTo(coinB.value))
+            .slice(
+              0,
+              Math.min(
+                looseAssets[key].length,
+                Math.floor(
+                  (byteLimit - getMessageSize(acc.newestInputs.length, acc.newOutputs.length) - newOutputSize) /
+                    newInputSize,
+                ),
+              ),
+            );
+          const newOutput: Output = {
+            asset: key,
+            value: newUnspents.reduce((left, right) => left.plus(right.value), new BigNumber('0')),
+            address: from.address,
+          };
+
+          return {
+            newestInputs: acc.newestInputs.concat(this.convertInputs(newUnspents)),
+            newOutputs: acc.newOutputs.concat(this.convertOutputs([newOutput])),
+          };
+        },
+        { newestInputs: newInputs, newOutputs: outputArray() },
+      );
+
+    switch (transactionUnsignedIn.type) {
```

Comment:
This should just call `transactionUnsigned.clone()` and let the subclasses deal with it.